### PR TITLE
Parse the new configuration for attribute keys

### PR DESCRIPTION
**Pull request by strategio**
https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-6606

### CODE REVIEW

File: src/class-wpml-gutenberg-strings-in-block.php [hunk 1]
```diff
@@ -214,8 +214,8 @@ private function get_block_queries( WP_Block_Parser_Block $block ) {
 			$this->block_types = $this->config_option->get();
 		}
 
-		if ( isset( $block->blockName, $block->innerHTML ) && array_key_exists( $block->blockName, $this->block_types ) ) {
-			return $this->block_types[ $block->blockName ];
+		if ( isset( $block->blockName, $this->block_types[ $block->blockName ]['xpath'] ) ) {
```
Comment (strategio) [inline src/class-wpml-gutenberg-strings-in-block.php]: We were checking for `$block->innerHTML` here but we are inside the private method `get_block_queries`. This is not appropriate to me and I will reuse this method for the keys later.

So I moved the check where it's really needed (and covered with tests).

File: src/class-wpml-gutenberg-strings-in-block.php [hunk 2]
```diff
@@ -25,9 +25,9 @@ public function find( WP_Block_Parser_Block $block ) {
 
 		$block_queries = $this->get_block_queries( $block );
 
-		if ( is_array( $block_queries ) ) {
+		if ( is_array( $block_queries ) && isset( $block->innerHTML ) ) {
 
-			$xpath = $this->get_domxpath( $block );
+			$xpath = $this->get_domxpath( $block->innerHTML );
```
Comment (andreasciamanna) [inline src/class-wpml-gutenberg-strings-in-block.php]: Note for other reviewers: this change is explained by Pierre in the comment below

File: tests/phpunit/tests/test-wpml-gutenberg-config-option.php [hunk 3]
```diff
@@ -14,33 +14,163 @@ class Test_WPML_Gutenberg_Integration_Config_Option extends OTGS_TestCase {
...
-			)
+		                        array(
+			                        'times' => 1,
+			                        'args'  => array(
+				                        WPML_Gutenberg_Config_Option::OPTION,
+				                        array( $block_type => $expected_block_config ),
+			                        ),
+		                        ) );
+
+		$subject->update_from_config( $config_settings );
```
Comment (andreasciamanna) [inline tests/phpunit/tests/test-wpml-gutenberg-config-option.php:45]: @strategio maybe I'm just missing it, but where is `$config_settings` defined?

**Review by andreasciamanna — CHANGES_REQUESTED** [review]
@strategio Unless I've missed it, I think we don't have tests where we pass a real XML as input.

I would at least ad one or more entries to `\Test_WPML_XML_Config_Validate::dp_valid_xml` so to test the XML validation with the new cases.

File: tests/phpunit/tests/test-wpml-gutenberg-config-option.php (hunk 3 above)
Comment (strategio) [reply tests/phpunit/tests/test-wpml-gutenberg-config-option.php:45]: It's defined here https://github.com/OnTheGoSystems/wpml-page-builders-gutenberg/pull/39/files#diff-df6372bc952d21fce216cca2189ba47cR28 @andreasciamanna. The display of the changes is not really helping.

**Comment by strategio** [comment]
@andreasciamanna, we don't receive an XML input, but we get an array representation of this XML input provided by the filter `wpml_config_array`.

File: tests/phpunit/tests/test-wpml-gutenberg-config-option.php (hunk 3 above)
Comment (andreasciamanna) [reply tests/phpunit/tests/test-wpml-gutenberg-config-option.php:45]: 👍

**Comment by andreasciamanna** [comment]
> @andreasciamanna, we don't receive an XML input, but we get an array representation of this XML input provided by the filter `wpml_config_array`.

@strategio I don't understand your answer.

Why you can't add one or more entries to `\Test_WPML_XML_Config_Validate::dp_valid_xml`?

**Comment by strategio** [comment]
>Why you can't add one or more entries to \Test_WPML_XML_Config_Validate::dp_valid_xml?

Sorry @andreasciamanna, I didn't pay attention you were talking about another class in the Core. I will do this!

**Comment by strategio** [comment]
@andreasciamanna, I added the test here https://git.onthegosystems.com/wpml/sitepress-multilingual-cms/merge_requests/2384/diffs?commit_id=24002a1cf42a8b7e51ee1c5cf7d3d6c82851a246.

**Review by andreasciamanna — APPROVED** [review]

**Comment by strategio** [comment]
@OnTheGoSystems/wpml please one more review.

**Review by brucepearson — APPROVED** [review]
@strategio It looks good. +1
